### ISSUE TITLE
Release and re-upload failed due to outage

### ISSUE THREAD

**Issue by eggplants** [pypi/warehouse#17235]
**Describe the bug**

According to [Python Status page](https://status.python.org/), PyPI was facing the CDN and backend outage in 2024-12-06 1:10-1:45 JST. (2024-12-05 16:10-16:45)

In the duration, I was releasing some [openapi-generator-cli](https://pypi.org/project/openapi-generator-cli) versions with Actions: 
https://github.com/OpenAPITools/openapi-generator-pip/actions/runs/12183546881/workflow
https://github.com/OpenAPITools/openapi-generator-pip/actions/runs/12183546881/job/33985224778

After the outage I noticed that almost "released" versions are not released in actual and re-uploading cannot be accepted due to `400 Client Error: File already exists` error.

In short, it seems that an outage occurred in PyPI's CDN and Backend during the package release, so the tar.gz file was uploaded but not released, and that version cannot be released.

Is it possible to have their files removed from PyPI internally?

<details>

<summary><code>poetry run task publish</code> in my local env</summary>

`publish.py`:
https://github.com/OpenAPITools/openapi-generator-pip/blob/master/publish.py

```shellsession
$ poetry run task publish
[4.0.0-beta] Downloading...
[4.0.0-beta] URL: 'https://search.maven.org/remotecontent?filepath=org/openapitools/openapi-generator-cli/4.0.0-beta/openapi-generator-cli-4.0.0-beta.jar'
[4.0.0-beta] Updating package version...
[4.0.0-beta] Testing...
======================================================== test session starts =========================================================
platform linux -- Python 3.12.3, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/eggplants/prog/openapi-generator-pip
configfile: pyproject.toml
collected 3 items


========================================================= 3 passed in 1.44s ==========================================================
[4.0.0-beta] Building...
Using virtualenv: /home/eggplants/.cache/pypoetry/virtualenvs/openapi-generator-cli-XnyH23gs-py3.12
Building openapi-generator-cli (4.0.0-beta)
  - Building sdist
  - Built openapi_generator_cli-4.0.0b0.tar.gz
  - Building wheel
  - Built openapi_generator_cli-4.0.0b0-py3-none-any.whl
[4.0.0-beta] Publishing to TestPyPI...
[4.0.0-beta] Publishing to PyPI...

Publishing openapi-generator-cli (4.0.0-beta) to PyPI
 - Uploading openapi_generator_cli-4.0.0b0-py3-none-any.whl 100%
 - Uploading openapi_generator_cli-4.0.0b0.tar.gz FAILED

  HTTPError

  400 Client Error: File already exists ('openapi_generator_cli-4.0.0b0.tar.gz', with blake2_256 hash 'd7667a9175d02f486a91fd7d171bf8a5240e81f766b1f065f521840042c7398a'). See https://pypi.org/help/#file-name-reuse for more information. for url: https://upload.pypi.org/legacy/

  at ~/.local/share/pipx/venvs/poetry/lib/python3.12/site-packages/requests/models.py:1021 in raise_for_status
      1017│                 f"{self.status_code} Server Error: {reason} for url: {self.url}"
      1018│             )
      1019│
      1020│         if http_error_msg:
    → 1021│             raise HTTPError(http_error_msg, response=self)
      1022│
      1023│     def close(self):
      1024│         """Releases the connection back to the pool. Once this method has been
      1025│         called the underlying ``raw`` object must not be accessed again.

The following error occurred when trying to handle this error:


  UploadError

  HTTP Error 400: File already exists ('openapi_generator_cli-4.0.0b0.tar.gz', with blake2_256 hash 'd7667a9175d02f486a91fd7d171bf8a5240e81f766b1f065f521840042c7398a'). See https://pypi.org/help/#file-name-reuse for more information. | b"<html>\n <head>\n  <title>400 File already exists ('openapi_generator_cli-4.0.0b0.tar.gz', with blake2_256 hash 'd7667a9175d02f486a91fd7d171bf8a5240e81f766b1f065f521840042c7398a'). See https://pypi.org/help/#file-name-reuse for more information.\n \n <body>\n  <h1>400 File already exists ('openapi_generator_cli-4.0.0b0.tar.gz', with blake2_256 hash 'd7667a9175d02f486a91fd7d171bf8a5240e81f766b1f065f521840042c7398a'). See https://pypi.org/help/#file-name-reuse for more information.\n  The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>\nFile already exists (&#x27;openapi_generator_cli-4.0.0b0.tar.gz&#x27;, with blake2_256 hash &#x27;d7667a9175d02f486a91fd7d171bf8a5240e81f766b1f065f521840042c7398a&#x27;). See https://pypi.org/help/#file-name-reuse for more information.\n\n\n \n"

  at ~/.local/share/pipx/venvs/poetry/lib/python3.12/site-packages/poetry/publishing/uploader.py:284 in _upload_file
      280│                 if self._io.output.is_decorated():
      281│                     self._io.overwrite(
      282│                         f" - Uploading {file.name} FAILED"
      283│                     )
    → 284│                 raise UploadError(e)
      285│             finally:
      286│                 self._io.write_line("")
      287│
      288│     def _register(self, session: requests.Session, url: str) -> requests.Response:
Traceback (most recent call last):
  File "/home/eggplants/prog/openapi-generator-pip/publish.py", line 122, in <module>
    publish(dryrun=os.getenv("DRYRUN") == "1")
  File "/home/eggplants/prog/openapi-generator-pip/publish.py", line 113, in publish
    subprocess.check_call([poetry_path, "publish", "-v"])
  File "/home/eggplants/.pyenv/versions/3.12.3/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/eggplants/.local/bin/poetry', 'publish', '-v']' returned non-zero exit status 1.
```

</details>

**Expected behavior**
We are able to re-publishing unreleased versions.

**To Reproduce**
I don't know.

**My Platform**
local env:
```shellsession
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.1 LTS
Release:        24.04
Codename:       noble
$ uname -vorm
6.8.0-49-generic #49-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov  4 02:06:24 UTC 2024 x86_64 GNU/Linux
$ poetry --version
Poetry (version 1.8.4)
$ python -V
Python 3.12.3
```

**Additional context**
My PyPI account: https://pypi.org/user/eggplants/
Related: https://github.com/OpenAPITools/openapi-generator-pip/issues/16

**Comment by di** [comment]
I'm not sure I'm following, I see the file you're attempting to re-release is published here: https://pypi.org/project/openapi-generator-cli/4.0.0b0/#openapi_generator_cli-4.0.0b0.tar.gz

> After the outage I noticed that almost "released" versions are not released in actual

What is indicating to you that these versions are not actually released?

**Comment by eggplants** [comment]
If they have been released, can we wait a bit for them to appear?

**Comment by eggplants** [comment]
Does this mean that the release has been made, even though it is not displayed?

**Comment by di** [comment]
I see it at that url:

![Image](https://github.com/user-attachments/assets/ba4cb3e6-883b-4452-bf46-1dbeada63df3)

Can you show me what you are seeing?

**Comment by eggplants** [comment]
`curl -s 'https://pypi.org/pypi/openapi-generator-cli/json' | jq -r '.releases | keys[]'` returns all "released" versions and the version `3.0.2` not shown in the history page yet can be installed with `pipx`. It appears that the release had been completed, just not displayed.

Also, the history page now shows all the versions that were currently released, so this issue does not seem to be a problem. Sorry for the trouble!

<details>

```shellsession
$ curl -s 'https://pypi.org/pypi/openapi-generator-cli/json' | jq -r '.releases | keys[]'
3.0.0
3.0.1
3.0.2
3.0.3
3.1.0
3.1.1
3.1.2
3.2.0
3.2.1
3.2.2
3.2.3
3.3.0
3.3.1
3.3.2
3.3.3
3.3.4
4.0.0
4.0.0b0
4.0.0b2
4.0.0b3
4.0.1
4.0.2
4.0.3
4.1.0
4.1.1
4.1.2
4.1.3
4.2.0
4.2.1
4.2.2
4.2.3
4.3.0
4.3.1
4.3.1.post0
5.0.0
5.0.0b0
5.0.0b2
5.0.0b3
5.0.1
5.1.0
5.1.1
5.2.0
5.2.1
5.3.0
5.3.1
5.4.0
6.0.0
6.0.0b0
6.0.1
6.1.0
6.2.0
6.2.1
6.3.0
6.4.0
6.5.0
6.6.0
7.0.0
7.0.0b0
7.0.1
7.1.0
7.10.0
7.10.0.post0
7.2.0
7.3.0
7.4.0
7.5.0
7.6.0
7.7.0
7.8.0
7.9.0
$ pipx install openapi-generator-cli==3.0.0
  installed package openapi-generator-cli 3.0.0, installed using Python 3.12.3
  These apps are now globally available
    - openapi-generator-cli
done! ✨ 🌟 ✨
```

</details>

**Comment by eggplants** [comment]
Is there some time lag after the release? Or was the outage related? Either way, it looks fine now...
